gaps/gap-3_mid_agreement_payments/gap-3_mid_agreement_payments.md

```diff
@@ -11,30 +11,39 @@ type: Feature
+
+This specification describes the semantics of the required payment parameters and how Requestor and Provider should agree on them.
+
 
 ### New properties
 
 | Name | Type | Description |
 | ---- | ---- | ----------- |
-| `golem.com.scheme.payu.debit-note-interval-sec?` | Number (uint32) [Negotiable] | Minimum time (in seconds) between subsequent Debit Notes. The Provider must not issue the next Debit Note before the interval period. The Requestor ignores Debit Notes issued before interval elapses. |
-| `golem.com.scheme.payu.payment-timeout-sec?` | Number (uint32) [Negotiable] | Maximum time to receive payment for any Debit Note. This time is counted from Debit Note’s issue date. |
+| `golem.com.scheme.payu.debit-note.interval-sec?` | Number (uint32) [Negotiable] | Minimum time (in seconds) between subsequent Debit Notes related to a single Activity. The Provider must not issue the next Debit Note before the interval period. The Requestor ignores Debit Notes issued before interval elapses and terminates the offending agreement. |
+| `golem.com.scheme.payu.payment-timeout-sec?` | Number (uint32) [Negotiable] | Maximum time to receive payment for any Debit Note. This time is counted from Debit Note’s issue date. At the same time, this is the minimum interval (in seconds) between two subsequent payable Debit Notes (Debit Notes with `payment_due_date` property set) related to a single Activity. The Provider must not issue a next payable Debit Note before this interval elapses. The Requestor ignores any offending Debit Notes and terminates the agreement. |
 
 - Requestor can ignore Debit Note, if it came before expected interval
-  - If Requestor is spammed with Debit Notes by Provider, he is allowed to break Agreement stating (in TerminateAgreement) that the **Reason** for termination is Provider's perceived misconduct. He should set `golem.requestor.code` field in **Reason** structure to `TooManyDebitNotes`.
-- In the mechanism suggested above a Debit Note shall have PaymentDueDate set, and the values must be set so that they correspond with the agreed `payment-interval-sec` value.
-  - Intervals are relative to the agreement timestamp
+  - If the Requestor receives Debit Notes more often than the interval specified by the agreed `golem.com.scheme.payu.debit-note.interval-sec?`, they are allowed to break the Agreement stating (in TerminateAgreement) that the **Reason** for termination is Provider's perceived misconduct. They should set the `golem.requestor.code` field in **Reason** structure to `TooManyDebitNotes`.
+  - Normally, the sent Debit Notes serve as an information of the accumulated costs and as a keep-alive message to ensure both sides that the agreement is still in action. 
+- To enable mid-agreement (mid-activity) payments, the Provider may periodically send payable Debit Notes (Debit Notes that have the `payment_due_date` attribute set to a non-empty value) to the requestor.
```

Comment:
To enable mid-agreement payments, values of `payment-timeout-sec` and `interval-sec` have to be negotiated by both parties. Provider can enforce these negotiations if the declared computation time is longer than some Provider's internal value (10h by default); this behavior breaks backward compatibility. 

Debit notes with `payment_due_date` serve as a trigger for payment. _Payable_ debit notes do not have to be sent periodically, they just cannot be sent more often than `payment-timeout-sec`

Reply:
ah, yeah, my wording is confusing here - I meant "enable" as in "make use of", "open the possibility" and not "trigger", "turn on" -> I'll change the wording to less confusing :) 

good catch 👍

Reply:
fixed